### PR TITLE
RELATED: RAIL-4342 Optimisation of connecting attributes matrix loading

### PR DESCRIPTION
**Pull request by plavemar**
JIRA: RAIL-4342

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8672/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10528/ : FAILURE in 8m 56s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8673/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10529/ : FAILURE in 10m 38s

File: libs/sdk-ui-dashboard/src/model/commandHandlers/dashboard/initializeDashboardHandler/loadConnectingAttributesMatrix.ts [hunk 1]
```diff
@@ -82,7 +85,7 @@ export function* loadConnectingAttributesMatrix(
             connectingAttributeMatrix[column] = connectingAttributeMatrix[column]
                 ? connectingAttributeMatrix[column]
                 : [];
-            const connectingAttributes = connectingAttributeRefs[connectingAttributeIndex++].map((ref) =>
+            const connectingAttributes = connectingAttributeRefs[connectingAttributeIndex++]?.map((ref) =>
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/commandHandlers/dashboard/initializeDashboardHandler/loadConnectingAttributesMatrix.ts:88]: This is sketchy, do we want to increment the counter even if the given index is undefined?

Comment (plavemar) [reply libs/sdk-ui-dashboard/src/model/commandHandlers/dashboard/initializeDashboardHandler/loadConnectingAttributesMatrix.ts:88]: Yes, there could be items in `connectingAttributeRefs` after the undefined one for which we might need to get data

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8674/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10530/ : SUCCESS in 11m 20s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2718

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2495/ : SUCCESS in 1m 54s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2791/ : SUCCESS in 38s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1619/ : SUCCESS in 17m 39s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10531/ : SUCCESS in 11m 49s

**Pull request merged**